packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -1002,27 +1001,51 @@ contract Deploy is Deployer {
                 revert("Cannon prestate dump not found, generate it with `make cannon-prestate` in the monorepo root.");
             }
             commands[2] = string.concat("cat ", filePath, " | jq -r .pre");
-            mipsAbsolutePrestate = Claim.wrap(abi.decode(vm.ffi(commands), (bytes32)));
+            mipsAbsolutePrestate_ = Claim.wrap(abi.decode(vm.ffi(commands), (bytes32)));
             console.log(
                 "[Cannon Dispute Game] Using devnet MIPS Absolute prestate: %s",
-                vm.toString(Claim.unwrap(mipsAbsolutePrestate))
+                vm.toString(Claim.unwrap(mipsAbsolutePrestate_))
             );
         } else {
             console.log(
                 "[Cannon Dispute Game] Using absolute prestate from config: %s", cfg.faultGameAbsolutePrestate()
             );
-            mipsAbsolutePrestate = Claim.wrap(bytes32(cfg.faultGameAbsolutePrestate()));
+            mipsAbsolutePrestate_ = Claim.wrap(bytes32(cfg.faultGameAbsolutePrestate()));
         }
+    }
+
+    /// @notice Sets the implementation for the `FAULT` game type in the `DisputeGameFactory`
+    function setCannonFaultGameImplementation() public onlyDevnet broadcast {
+        console.log("Setting Cannon FaultDisputeGame implementation");
+        DisputeGameFactory factory = DisputeGameFactory(mustGetAddress("DisputeGameFactoryProxy"));
 
         // Set the Cannon FaultDisputeGame implementation in the factory.
-        _setFaultGameImplementation(
-            factory, GameTypes.FAULT, mipsAbsolutePrestate, IBigStepper(mustGetAddress("Mips")), cfg.faultGameMaxDepth()
-        );
+        _setFaultGameImplementation({
+            _factory: factory,
+            _gameType: GameTypes.FAULT,
+            _absolutePrestate: loadMipsAbsolutePrestate(),
+            _faultVm: IBigStepper(mustGetAddress("Mips")),
+            _maxGameDepth: cfg.faultGameMaxDepth()
+        });
+    }
+
+    /// @notice Sets the implementation for the cannon output bisection game type (253) in the `DisputeGameFactory`
+    function setCannonOutputBisectionGameImplementation() public onlyDevnet broadcast {
+        console.log("Setting Cannon OutputBisectionGame implementation");
+        DisputeGameFactory factory = DisputeGameFactory(mustGetAddress("DisputeGameFactoryProxy"));
+
+        _setFaultGameImplementation({
+            _factory: factory,
+            _gameType: GameType.wrap(253),
```

Comment:
Not sure, was following standard here. @clabby ?

Reply:
I'd be tempted to either remove the enums or make them more like the trace types in cannon that are quite specific (ie `FAULT` is too generic).  Would need to encapsulate both the VM and the FPP in use, so maybe `OP_CANNON`?  The current game type 0 that's just cannon maybe just becomes `FPA_CANNON`.

Reply:
Oh but can we make this game type `1` rather than `253`?   This will be the new primary game type deployed everywhere so doesn't need to be a high number.  My suggestion is that we stick with game type 1 permanently and just drop game type 0 like we would if we'd deployed type 0 to mainnet.